test/e2e/integration_test.go

```diff
@@ -0,0 +1,116 @@
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"net/http"
+	// "k8s.io/utils/pointer"
+
+	capi_e2e "sigs.k8s.io/cluster-api/test/e2e"
+	// "sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	// "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	k8sVersion          string
+	containerRuntime string
+	err                 error
+)
+
+var _ = Describe("When testing integration [integration]", func() {
+	BeforeEach(func() {
+		initVars()
+	})
+	It("CI Test Provision", func() {
+		// ciTestProvision()
+		healthCheck()
+	})
+})
+
+// Get variable values from env
+func initVars() {
+	Expect(err).NotTo(HaveOccurred())
+	k8sVersion = os.Getenv("KUBERNETES_VERSION")
+	ironicContainers = []string{
+		"ironic",
+		"ironic-inspector",
+		"ironic-endpoint-keepalived",
+		"ironic-log-watch",
+		"dnsmasq",
+	}
+	generalContainers = []string{
+		"httpd-infra",
+		"registry",
+		"sushy-tools",
+		"vbmc",
+	}
+	containerRuntime = e2eConfig.GetVariable("CONTAINER_RUNTIME")
+}
+
+func ciTestProvision() {
+	By("Provision Workload cluster")
+	createTargetCluster(k8sVersion)
+	By("Download and Install Calico")
+	cniPath := e2eConfig.GetVariable(capi_e2e.CNIPath)
+	if osType == "centos" {
+		updateCalico(e2eConfig, cniPath, "eth1")
+	} else {
+		updateCalico(e2eConfig, cniPath, "enp2s0")
+	}
+	By("Verify Calico pods running")
+	targetCluster = bootstrapClusterProxy.GetWorkloadCluster(ctx, namespace, clusterName)
+	targetClusterClientset := targetCluster.GetClientSet()
+	waitForAllPodsRunning(targetClusterClientset, "kube-system", 9, 20)
+	By("Wait for all pods to be in running state")
+	waitForAllPodsRunning(targetClusterClientset, namespace, 150, 20)
+	By("Wait for all nodes to be in ready state")
+	waitForAllNodesReady(targetClusterClientset, 150, 3)
+	By("Pivot objects to target cluster")
+	pivoting(ctx, func() PivotingInput {
+		return PivotingInput{
+			E2EConfig:             e2eConfig,
+			BootstrapClusterProxy: bootstrapClusterProxy,
+			TargetCluster:         targetCluster,
+			SpecName:              specName,
+			ClusterName:           clusterName,
+			Namespace:             namespace,
+			ArtifactFolder:        artifactFolder,
+			ClusterctlConfigPath:  clusterctlConfigPath,
+		}
+	})
+	By("Expect 401 from /v1/nodes endpoint")
+	ironicNodeEndpoint := fmt.Sprintf("%snodes", e2eConfig.GetVariable("IRONIC_URL"))
+	_, err := http.Get(ironicNodeEndpoint)
+	Expect(err).To(HaveOccurred())
+	By("Repivot everything back to the source clusternil")
+	rePivoting(ctx, func() RePivotingInput {
+		return RePivotingInput{
+			E2EConfig:             e2eConfig,
+			BootstrapClusterProxy: bootstrapClusterProxy,
+			TargetCluster:         targetCluster,
+			SpecName:              specName,
+			ClusterName:           clusterName,
+			Namespace:             namespace,
+			ArtifactFolder:        artifactFolder,
+			ClusterctlConfigPath:  clusterctlConfigPath,
+		}
+	})
+}
+
+func healthCheck() {
```

Comment:
What tasks from ansible integration you are trying to implement here I do not see healthcheck tasks in integration logs at least